app/services/auth.server.ts

```diff
@@ -0,0 +1,49 @@
+import { User } from "@prisma/client";
+import { Authenticator } from "remix-auth";
+import { GitHubStrategy } from "remix-auth-github";
+import { GoogleStrategy } from "remix-auth-google";
+import { sessionStorage } from "~/services/session.server";
+import {
+  createOrLoginWithGithub,
+  createOrLoginWithGoogle,
+} from "~/shared/db/user.server";
+
+const url = `${
+  process.env.VERCEL_URL
+    ? `https://${process.env.VERCEL_URL}`
+    : "http://localhost:3000"
+}/auth`;
+
+const github = new GitHubStrategy(
+  {
+    clientID: process.env.GH_CLIENT_ID as string,
+    clientSecret: process.env.GH_CLIENT_SECRET as string,
+    callbackURL: `${url}/github/callback`,
+  },
+  async ({ profile, context }) => {
+    const user = createOrLoginWithGithub(profile, context?.userId);
```

Comment:
missing await? wonder how did it work if the function is async

Reply:
Oh I guess that works because you return a promise to an async function so it resolves it later, but still better to await here for clarity?

Reply:
yap! Done!